### ISSUE TITLE
demo.py

### ISSUE THREAD

**Issue by rasemailcz** [naver/croco#35]
Dears,

I need help with understanding the demo.py script.

I thought that the pre-training task is: "reconstruct masked areas from an image, given a different view of the same scene and a partially masked image".

Now, on lines 33-36, each patch in the output of the network is "put back to RGB space". However, by doing this "normalization", which is the term used in the comment, the mean and std of each patch, both "not masked" AND "masked", is set.

Therefore, the pre-training task appears to be better described by: "reconstruct masked areas from an image, given a different view of the same scene and a partially masked image, and if you don't know, we'll help you by peaking at your test data and fill in mean and standard deviation, at least".

Am I wrong?

Of course, I tried to mask the input image "properly" and I attach the output on the demo data, which appears that the network fails on the pre-train task completely...

![demo_output](https://github.com/user-attachments/assets/1ee63d99-39a2-48ae-a9f6-a5051e2a3f24)

EDIT: 
I understand the problem now. 

I modified the lines 34-35 to
```
mean = patchified[~mask].mean(dim=(-1, -2), keepdim=True)
var = patchified[~mask].var(dim=(-1, -2), keepdim=True)
```
- do you agree that this is the correct approach? This version produces: ![demo_output](https://github.com/user-attachments/assets/ce6514ff-debb-4818-82b6-610ae346e6eb)

**Comment by PhilippeWeinzaepfel** [comment]
Hi,

We had both pretrained models where the loss is applied directly on RGB predictions or on predictions after normalization according to mean and std over each patch.Confirming what have been found in eg MAE, we obtain slightly better pre-trained models when applying this normalization, and we thus release that ones.

While models that predict values in RGB spaces can be directly visualized, human eyes are not great at visualizing normalized values. For visualization purposes, we thus undo this normalization in the demo. An alternative to visualize the output of the pretext task would be to display the target & prediction directly but this won't lie in the RGB space directly, so you cannot directly do as you seem to have done.

Note again that our goal is to learn a good pre-training, not to predict nice completion results, in which case a GAN, a diffusion model or any generative model would clearly do a much better job.

Best
